RELEASE_NOTES.md

```diff
@@ -20,6 +20,58 @@ Notable changes include:
   * Bug fixes/improvements:
 
 
+Version 2024.07.0 -- Release date 2024-07-xx
+============================================
+
+This release contains new features, improvements, and bugfixes.
+
+Notable changes include:
+
+  * New features / API changes:
+     * Added support for a "multi-reduction" operation which allows users to
+       perform a run time-defined number of reduction operations in a kernel.
+       Please see the RAJA User Guide for details and examples.
+     * Added first couple of sections for a "RAJA Cookbook" in the RAJA User
+       Guide. The goal is to provide users with more detailed guidance about
+       using RAJA features, choosing execution policies, etc. Additional
+       content will be provided in future releases.
+     * Added atomicLoad and atomicStore routines for correctness in some
+       use cases.
+     * Added OpenMP 5.1 implementations for atomicMin and atomicMax.
+     * Add SYCL reduction support in RAJA::launch
+
+  * Build changes/improvements:
+     * Update camp submodule to v2024.07.0 release. This will be a version
+       constraint for this release in RAJA Spack package.
+
+  * Bug fixes/improvements:
+     * Fix CMake issue for case when RAJA is used as a submodule dependency.
+     * Various fixes and improvements to builtin atomic support.
+     * Fixes and improvements to other atomic operations:
+        * Modified HIP and CUDA generic atomic compare and swap algorithms
+          to use atomic loads instead of relying on volatile.
+        * Re-implemented atomic loads in terms of builtin atomics for CUDA
+          and HIP so that the generic compare and swap functions can use it.
+        * Removes volatile qualifier in atomic function signatures.
+        * Use cuda::atomic_ref in newer versions of CUDA to back 
+          atomicLoad/atomicStore.
+        * Use atomicAdd as a fallback for atomicSub in CUDA.
+        * Removed checks where __CUDA_ARCH__ is less than 350 since RAJA 
+          requires that as the minimum supported architecture (CMake check).
+     * Fix issue related to the fact that only C version of memcpy is supported
```

Comment:
This was part of the churn for atomics, so I'm not sure it needs to be mentioned (it wasn't an outstanding bug from the last release).

Reply:
The cuda arch check?

Reply:
No, the C version of memcpy

Reply:
Gotcha. I will remove from release notes.